components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/consumer/KinesisDefaultResumeAdapter.java

```diff
@@ -46,8 +47,8 @@ public void setRequestBuilder(GetShardIteratorRequest.Builder resumable) {
 
     @Override
     public void resume() {
-        assert streamName != null;
-        assert resumable != null;
```

Comment:
Using Java's assert is the correct away to assert for nullity in non-public code as is the case here (obs.: the code is internally public, but not public in the to Camel users).

Reply:
Thanks for your feedback.

IMHO no matter if it's about a user-facing pubic or not-public logic, at the end of the day this code would run in a production environment where  typically Java assertion is not explicitly enabled. Then having these null guards using Java assertion would simply get skipped, then users would only see the **side effects** of these null values **later on** on their call stack and they / we would need to guess and find out the **root cause** (which was null values not being checked early enough). On the other hand having this check eagerly through `ObjectHelper#notNull` would not end up in such a scenarios.

Reply:
These 2 are [invariants which is why checking for assertions is the correct approach](https://docs.oracle.com/javase/7/docs/technotes/guides/language/assert.html#usage-invariants). In normal circumstances they should be defined as final members of the `KinesisDefaultResumeAdapter` class. Unfortunately our code is made in a way that does not favor usage of final variables in many places (which then, would make the assert unnecessary)

Reply:
> These 2 are [invariants which is why checking for assertions is the correct approach](https://docs.oracle.com/javase/7/docs/technotes/guides/language/assert.html#usage-invariants). In normal circumstances they should be defined as final members of the `KinesisDefaultResumeAdapter` class. Unfortunately our code is made in a way that does not favor usage of final variables in many places (which then, would make the assert unnecessary)

In the very same page [it says](https://docs.oracle.com/javase/7/docs/technotes/guides/language/assert.html#putting-assertions-into-your-code):

> Do not use assertions for argument checking in public methods.
Argument checking is typically part of the published specifications (or contract) of a method, and these specifications must be obeyed whether assertions are enabled or disabled. Another problem with using assertions for argument checking is that erroneous arguments should result in an appropriate runtime exception (such as IllegalArgumentException, IndexOutOfBoundsException, or NullPointerException). An assertion failure will not throw an appropriate exception.

And by Camel we additionally make a differentiation of about if that given public method is a _user-facing public code_ or not and then we apply the assert approach or not depending on that?  As a concrete example consider the change by *this* conversation here. Even if this class is not a _user-facing public code_, according to the recommendation :point_up_2: that assert is done inside it's contract implementation method, namely `org.apache.camel.resume.ResumeAdapter#resume` which is wrong according to the recommendation :point_up_2:.

```
@Override
    public void resume() {
        assert streamName != null;
        assert resumable != null;
       ...
       ...
}
```

But I will apply the changes the way you are asking me to do, it's alright 👍